File: Builder/Ruby/ExcelReportBuilder.rb
require_relative "Report"
require_relative "IReportBuilder"

class ExcelReportBuilder < IReportBuilder
    def buildContent
        if !@report
            @report = Report.new
        end

        @report.content = "Tables"
    end

    def buildFile
        if !@report
            @report = Report.new
        end

        @report.filename = "Report.xlsx"
    end

    def getResult
        @report
    end
end

File: Builder/Javascript/ExcelReportBuilder.js
import IReportBuilder from "./IReportBuilder.js";
import Report from "./Report.js";

class ExcelReportBuilder extends IReportBuilder {
  buildContent() {
    if (!this.report) {
      this.report = new Report();
    }

    this.report.content = "Tables";
  }

  buildFile() {
    if (!this.report) {
      this.report = new Report();
    }

    this.report.filename = "Report.xlsx";
  }

  getResult() {
    return this.report;
  }
}

export default ExcelReportBuilder;